go/libkb/session.go

```diff
@@ -285,13 +287,29 @@ func (s *Session) check() error {
 		}
 	} else {
 		s.G().Log.Notice("Stored session expired")
-		s.valid = false
+		s.Invalidate()
 	}
 
 	s.G().Log.Debug("- Checked session")
 	return nil
 }
 
+// Invalidate marks the session as invalid and posts a logout
+// notification.
+func (s *Session) Invalidate() {
+	s.G().Log.Debug("invalidating session")
+	s.clearState()
+	s.G().NotifyRouter.HandleLogout()
```

Comment:
Should `clearState()` and `Invalidate` be the same function?  I vote yes.  And then we should get rid of the call to `HandleLogout` from `Account:Logout` and then we're only calling `HandleLogout` once.